### ISSUE TITLE
Required Security Functionalities

### ISSUE THREAD

**Issue by as3810t** [as3810t/COMPSEC-HW-GROTESQUE-GECKO#4]
[WIKI](https://github.com/as3810t/COMPSEC-HW-GROTESQUE-GECKO/wiki/Required-Security-Functionalities)

**Comment by as3810t** [comment]
Based on the online meeting, the following decisions were made:

- Communication
  - HTTPS
- Passwords: hashing, salting and policy
  - Salting: cryptographically secure random generator
  - Hashing: PBKDF2 (with cryptographically secure hash function)
  - Policy
    - Minimum length: 8
- Authentication
  - Password based
  - Password reset (email needed)
- Session
  - JWT (JSON Web Token): signed with asymmetric key
- Database
  - Server and client side certificates
- DOS attack
  - Rate limiting

**Comment by sarvaryk** [comment]
We wrote some lines about the Required Security Functionalities. If you (@as3810t @futobarna ) have any idea for suplementing the document, then please let me know.

**Comment by futobarna** [comment]
I wouldn't add anything to this, my only small nitpick is that DoS in this context is usually written with a lowercase o.

**Comment by as3810t** [comment]
I would rephrase the database passage to be a bit more precise:

> To mitigate impersonation, the database connection will be verified by a certificate, and the server will login to the MySQL server using a client side certificate.

Otherwise I think it is OK.

**Comment by sarvaryk** [comment]
Thank you for your feedbacks! The suggestions are now reflected in the document.
